cpu/sam0_common/periph/adc.c

```diff
@@ -346,30 +442,46 @@ static Adc *_adc(uint8_t dev)
 #endif
 }
 
-static int32_t _sample(adc_t line)
+static inline void _config_line(Adc *dev, adc_t line, bool diffmode, bool freerun,
+                                uint8_t samplen)
 {
-    Adc *dev = _dev(line);
-    bool diffmode = adc_channels[line].inputctrl & ADC_INPUTCTRL_DIFFMODE;
-
+    dev->SAMPCTRL.reg = samplen;
     dev->INPUTCTRL.reg = ADC_GAIN_FACTOR_DEFAULT
                        | adc_channels[line].inputctrl
                        | (diffmode ? 0 : ADC_NEG_INPUT);
-#ifdef ADC_CTRLB_DIFFMODE
     if (diffmode) {
+#if   defined(ADC_CTRLB_DIFFMODE)
```

Comment:
```suggestion
#if defined(ADC_CTRLB_DIFFMODE)
```

I think the indentation is confusing here.